src/components/NewCommentForm.tsx

```
@@ -1,8 +1,82 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import PropTypes from 'prop-types';
+
+type Props = {
+  onSubmit: (comment: Omit<Comment, 'id'>) => Promise<void>;
+  selectedPost?: Post;
+};
+
+type FormState = {
+  name: string;
+  email: string;
+  body: string;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ onSubmit, selectedPost }) => {
+  const [form, setForm] = useState<FormState>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [errors, setErrors] = useState<FormState>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  function handleChange<K extends keyof FormState>(key: K, value: string) {
+    setForm(prev => ({ ...prev, [key]: value }));
+    setErrors(prev => ({ ...prev, [key]: '' }));
+  }
+
+  function clearForm() {
+    setForm({ name: '', email: '', body: '' });
+    setErrors({ name: '', email: '', body: '' });
+    setIsLoading(false);
+  }
+
+  function resetBody() {
+    setForm(prev => ({ ...prev, body: '' }));
+    setErrors(prev => ({ ...prev, body: '' }));
+  }
+
+  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+
+    if (!selectedPost?.id) {
+      return;
+    }
+
+    const newErrors: FormState = {
+      name: form.name.trim() ? '' : 'Name is required',
+      email: form.email.trim() ? '' : 'Email is required',
+      body: form.body.trim() ? '' : 'Comment is required',
+    };
+
+    setErrors(newErrors);
+
+    if (Object.values(newErrors).some(Boolean)) {
+      return;
+    }
+
+    try {
+      setIsLoading(true);
+      await onSubmit({ ...form, postId: selectedPost.id });
+      resetBody();
+    } catch {
```

The catch block is empty, which swallows any exceptions thrown by `onSubmit`. This makes debugging and error handling harder and can hide failures from tests or users. Prefer at least logging the error or setting an error state so the UI (or parent) can react.

Suggested approaches:
- Log the error: `catch (err) { console.error(err); }`
- Or set a submission error state and show a notification so the user can retry.

This follows the checklist guidance to use try/catch for async error handling (optional): "Prefer try { .. } catch for async error handling."